NerdyPy/modules/league.py

```diff
@@ -57,7 +57,7 @@ async def summoner(self, ctx, region: RegionConverter, *, summoner_name: str):
         """get information about the summoner"""
         rank = tier = lp = wins = losses = ""
 
-        auth_header = {"X-Riot-Token": config.riot}
+        auth_header = {"X-Riot-Token": self.config["riot"]}
```

Comment:
was würde an diesen stellen passieren wenn das riot segment nicht vorhanden ist?

Reply:
Das wirft nen KeyError
Das trifft hauptsächlich auf die .get und .set Funktionen zu, aber you get the point -> https://docs.python.org/3/library/configparser.html#exceptions

Reply:
deswegen hatte ich wegen der Modularität gefragt, das würde das ja verhindern und man könnte dann vorher ne Validierung der config machen

Reply:
Ich bin mir nicht sicher, wie wir das prüfen wollen.
Nicht jedes Modul braucht zwingend eine Config. Wenn ich einfach nur nen dummen Check mache, der prüft ob es eine Section in der Config mit dem Namen des Moduls gibt, dann lädt er keine Module, die keine Config brauchen.

Reply:
Ich weiß gerade nicht an welcher stelle die Cogs geladen werden, aber jedes Cog hat ne eigene setup methode mit dem bot. dort können dann indivituelle config validierungs checks gemacht werden.